lib/internal/url.js

```diff
@@ -753,13 +753,13 @@ ObjectDefineProperties(URLSearchParams.prototype, {
  * We use `href` and `protocol` as they are the only properties that are
  * easy to retrieve and calculate due to the lazy nature of the getters.
  *
- * We check for auth attribute to distinguish legacy url instance with
+ * We check for `auth` and `path` attribute to distinguish legacy url instance with
  * WHATWG URL instance.
  * @param {*} self
  * @returns {self is URL}
  */
 function isURL(self) {
-  return Boolean(self?.href && self.protocol && self.auth === undefined);
+  return Boolean(self?.href && self.protocol && self.auth === undefined && self.path === undefined);
```

Comment:
TBH, I don't think this is the ideal way to fix the issue https://github.com/nodejs/node/issues/48921 , but since `path` is a field in legacy URL object instance, and other issue has been fixed in same way: https://github.com/nodejs/node/pull/47886/files I think this fix maybe fine. If you have a better way, please let me know.